src/java.base/share/classes/java/lang/foreign/Linker.java

```diff
@@ -303,5 +311,67 @@ default MethodHandle downcallHandle(MemorySegment symbol, FunctionDescriptor fun
         static Option firstVariadicArg(int index) {
             return new LinkerOptions.FirstVariadicArg(index);
         }
+
+        /**
+         * {@return A linker option used to preserve values that might be overwritten by the runtime
+         *          before they can be read through conventional means}
+         * <p>
+         * A downcall method handle linked with this option will feature an additional {@link MemorySegment}
```

Comment:
We should say somewhere that the additional segment should be a native segment (right?)

Reply:
I've changed the text a bit to specify that a native segment should be used.

Reply:
Should we add a test for this?

Reply:
It's already covered by `TestPassHeapSegment` since we use the same code everywhere to unbox these segments.

I don't mind adding an explicit test for this case as well though.